### DISCUSSION TITLE
Why doesn't my Next + Prisma + Postgres blog update without a redeploy?

### DISCUSSION THREAD

**Discussion by zuff-digital** [prisma/prisma#12816] [Q&A] [answered]
I created a simple fullstack blog by following [this tutorial](https://vercel.com/guides/nextjs-prisma-postgres#step-1:-set-up-your-next.js-starter-project).

I'm finding that I can create new posts successfully via the UI (I can see them persisted in my Heroku-hosted Postgres db using `psql`), but they don't show up in the Public Feed section of the app unless I redeploy the app completely. 

Any idea what could be happening here?

**Comment by nurul3101** [comment]
Hey @zuff-digital 👋 
If I understand it correctly, you are able to create posts via Prisma Studio and it gets persisted in the PostgreSQL database but the posts don't appear in the application automatically, right?

**Reply by zuff-digital** [reply]
Hi @nurul3101! Yes that's correct, although by 'create posts through the UI' I meant directly through the frontend of the app itself (I believe using Prisma Studio should work as well, though I haven't gotten to test that). 

Is there any more information I can give you that's helpful? It's unclear to me exactly what could be happening, I followed the tutorial pretty closely and don't think I'm doing anything differently.

**Comment by austincrim** [comment, marked as answer]
@zuff-digital I believe this is because [step 4](https://vercel.com/guides/nextjs-prisma-postgres#step-4.-update-the-existing-views-to-load-data-from-the-database) of the tutorial uses `getStaticProps` to fetch the public feed data and that method only runs on build. The individual pages use `getServerSideProps` which run on each request and return data dynamically. You can read more about the difference [here](https://nextjs.org/docs/basic-features/data-fetching/overview).

If you want the public feed to update dynamically on request, change the data fetching method to `getServerSideProps`.

**Reply by zuff-digital** [reply]
Thanks a lot @austincrim, this seems to have done the trick! Do you suggest I open an issue or PR for this?

**Reply by austincrim** [reply]
We are going to update it on our end! Thanks so much for bringing it to our attention.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
